internal/svcmanager/svcmanager.go

```diff
@@ -97,3 +97,16 @@ func (m *Manager) Ready() bool {
 
 	return true
 }
+
+func (m *Manager) Model(ctx context.Context) (*model.SvcModel, error) {
+	if err := m.Wait(); err != nil {
+		return nil, errs.Wrap(err, "Wait failed before initializing svc model.")
+	}
+
+	svc, err := model.NewSvcModel(ctx, m.cfg)
+	if err != nil {
+		return nil, errs.Wrap(err, "Failed to initialize svc model.")
+
+	}
+	return svc, nil
+}
```

Comment:
Yeah, I agree. It just created circular dependencies, because the svcmanager, uses the svcmodel to ping the server.  But maybe it is still nicer if I use an interface, maybe named`ConnectionWaiter` here...

Reply:
I see what you're saying, but that feels like it's just covering up the real need. At first glance it's: "The application needs a service model to talk to the API. The service model needs a service manager before it can talk to the API. The service manager needs a service model to talk to the API." However, it is really: "The application needs a service model with a service manager to talk to the API. The service manager needs to talk to the API.".

Please consider if the service manager can have accessing this one endpoint (whatever the ping endpoint is) coded within it as a sort of bootstrap mechanism.

Reply:
I pushed a new version, where the svcManager is requested by the model constructor.  But, honestly, this is awkward in a completely different way, because now the constructor has this implicit logic to wait for the connection to become ready...

> edit to add: It may even make sense for only the service manager to have the ping endpoint in it. It's probably unlikely any normal app logic needs to call that directly. And, if so, the service model can expose that logic from the manager it contains.
The "ping" is just a request to return the State Tool version, so it cannot only live in the manager.

Reply:
I do not like the idea of duplicating the call in the manager, but I have a solution up that makes the circular dependency more explicit IMO.

Now, the svcmodel is also used to stop the service.  I missed that before, but I feel strongly that this should be a model.SvcModel function that should also be guarded by the svcmanager.Wait() function.

Reply:
nvmd.  I think you are right: The ping is now duplicated in the svcmanager.  Looks much cleaner.